vulcanai2/models/basenetwork.py

```diff
@@ -89,14 +96,46 @@ def __init__(self, name, dimensions, config, save_path=None, input_network=None,
             validation_accuracy=[]
         )
 
-        # self._itr = 0 #TODO: ?
+        self.in_dim = self._dimensions
+        if self._input_networks:
+            for i, in_net in enumerate(self._input_networks):
+                if in_net.__class__.__name__ == "ConvNet":
+                    if in_net.conv_flat_dim != self.in_dim:
+                        self.in_dim[i] = self.get_flattened_size(in_net)
+                    else:
+                        pass
 
+                if in_net.__class__.__name__ == "DenseNet":
+                    if in_net.dims[-1] != self.in_dim:
+                        self.in_dim = in_net.dims[-1]
+                    else:
+                        pass
         self._create_network(
             activation=activation,
             pred_activation=pred_activation)
 
     # TODO: where to do typechecking... just let everything fail?
 
+    @abc.abstractmethod
+    def _forward(self):
+        pass
+
+    def forward(self, inputs):
+        """Perform a forward pass through the module/modules."""
+
+        if self._input_networks:
+            if not isinstance(inputs, list):
+                inputs = [inputs]
+            models=self._input_networks
+            
+            outputs = []
+            for model, x in zip(models, inputs):
+                outputs.append(model(x))
+            network_output = self._forward(torch.cat(outputs, 1))
```

Comment:
why are you using `forward`? Also please specify the function inputs explicitly so the code is more readable. (e.g. what does the 1 mean? probably the axis on which to concat but it's best to specify each attribute explicitly)